src/figuresClasses.ts

```diff
@@ -1,19 +1,77 @@
-export interface Figure {
-
+enum Shape {
+  triangle = 'triangle',
+  circle = 'circle',
+  rectangle = 'rectangle',
 }
 
-export class Triangle {
+type Color = 'red' | 'green' | 'blue';
 
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
 }
 
-export class Circle {
+export class Triangle implements Figure {
+  public shape: Shape = Shape.triangle;
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (c >= a + b || b >= a + c || a >= b + c) {
+      throw new Error('Not a triangle');
+    }
+
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('The figure`s values should be more than 0');
+    }
+  }
 
+  getArea(): number {
+    const s = (this.a + this.b + this.c) / 2;
+    const aDiff = s - this.a;
+    const bDiff = s - this.b;
+    const cDiff = s - this.c;
+
+    return Math.round(Math.sqrt(s * aDiff * bDiff * cDiff) * 100) / 100;
+  }
 }
 
-export class Rectangle {
+export class Circle implements Figure {
+  public shape: Shape = Shape.circle;
+
+  constructor(public color: Color, public radius: number) {
+    if (radius <= 0) {
+      throw new Error('Error. The radius is less than 0');
+    }
+  }
 
+  getArea(): number {
+    return Math.floor(Math.PI * this.radius ** 2 * 100) / 100;
+  }
 }
 
-export function getInfo(figure) {
+export class Rectangle implements Figure {
+  public shape: Shape = Shape.rectangle;
+
+  constructor(
+    public color: Color,
+    public width: number,
+    public height: number,
+  ) {
+    if (width < 0 || height < 0) {
+      throw new Error('Error. One of sids is less than 0');
```

Comment:
Fix error description, width or height set 2 sides. 
so use  'Width or height length cannot be zero or less'
for example